File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int n=s.length(),maxLen=0,left=0,right=0;
        Set<Character>strSet=new HashSet<>();
        while(right<n){
            if(!strSet.contains(s.charAt(right))){
                strSet.add(s.charAt(right));
                maxLen=Math.max(maxLen,right-left+1);
                right++;
            }
            else{
                strSet.remove(s.charAt(left));
                left++;
            }
        }
        return maxLen;
    }
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        s1=[]
        maxi=0
        c=0
        for i in range(len(s)):
            if s[i] not in s1:
                s1.append(s[i])
                c+=1
                maxi=max(maxi,c)
            else:
                c=0
                s1.clear()
        return maxi